jira/jira-v1.asciidoc

```diff
@@ -0,0 +1,94 @@
+== Jira REST Configuration
+
+This documentation describes aspects of the Jira REST `jira-v1.json` file configuration such as the authentication methods, data crawled and retrieved, pagination information, variables used, and endpoints used. Terminology is also provided as a reference.
+
+The list of data the REST connector crawls using the jira REST configuration is:
+
+* Projects 
+* Comments
+* Worklogs  
+* Attachments
+
+
+== Authentication methods
+
+The Jira REST configuration supports:
+
+* Basic Authentication using the username and password from an Atlassian account. For more information, see link:https://developer.atlassian.com/cloud/confluence/basic-auth-for-rest-apis/[Basic auth for REST APIs | Atlassian Developer^].
+
+* API Token. For information about how to create a new API token, see link:https://id.atlassian.com/manage/api-tokens[API Tokens | Atlassian ID^].
+
+
+== Supported crawl options
+
+The Jira REST configuration supports the following crawl options:
+
+* Full crawl
+
+* Recrawl that uses the `rely on the strayContentDeletion` parameter
+
+
+== Pagination information
+
+This recipe is configured to use pagination by batch size. 
```

Comment:
This recipe is configured to use pagination by batch size. Needs to configure Query Params, and the property 'Pagination By BatchSize'
Query Params:
- `maxResults=${LW_BATCH_SIZE}`, where `${LW_BATCH_SIZE}` is a variable
- `startAt=${LW_INDEX_START}`, where `${LW_INDEX_START}` is a variable
2. Configure the 'Pagination By BatchSize' properties
- IndexStart: The starting point.  Set to 0 because the Atlassian pagination is zero-index-base. From the queries above, it replaces variable `${LW_INDEX_START}`,  the plugin internally increases the value of the indexStart ensuring each page is requested with an updated start-point.
- BatchSize: The number of elements to retrieve. Set to 50 by default. From the queries above, it replaces variable `${LW_BATCH_SIZE}`
- Stop Condition Key: Reference the “key” in the response,  that needs to be met in order to stop the pagination. To paginate issues, it must be “issues”, to paginate projects it must be 'values'
- Stop Condition Value: Reference the “value” in the response,  that needs to be met in order to stop the pagination. For the jira config, to stop pagination the list of objects retrieved must be empty, then the stop condition should be `[]`